lib/internal/process/per_thread.js

```diff
@@ -223,6 +229,49 @@ function wrapProcessMethods(binding) {
     return true;
   }
 
+  function replace(execPath, args, env) {
+    const { isMainThread } = require('internal/worker');
+
+    if (!isMainThread) {
+      throw new ERR_WORKER_UNSUPPORTED_OPERATION('Calling process.replace');
+    } else if (process.platform === 'win32') {
+      throw new ERR_PLATFORM_UNSUPPORTED_OPERATION('Calling process.replace');
+    }
+
+    validateString(execPath, 'execPath');
+    validateArray(args, 'args');
+
+    for (let i = 0; i < args.length; i++) {
+      const arg = args[i];
+      if (typeof arg !== 'string' || StringPrototypeIncludes(arg, '\u0000')) {
+        throw new ERR_INVALID_ARG_VALUE(`args[${i}]`, arg, 'must be a string without null bytes');
+      }
+    }
+
+    if (env !== undefined) {
+      validateObject(env, 'env');
+
+      for (const { 0: key, 1: value } of ObjectEntries(env)) {
+        if (
+          typeof key !== 'string' ||
+          typeof value !== 'string' ||
+          StringPrototypeIncludes(key, '\u0000') ||
+          StringPrototypeIncludes(value, '\u0000')
+        ) {
+          throw new ERR_INVALID_ARG_VALUE(
+            'env', env, 'must be an object with string keys and values without null bytes',
+          );
+        }
+      }
+    }
+
+    // Construct the environment array.
+    const envArray = ObjectEntries(env || {}).map(({ 0: key, 1: value }) => `${key}=${value}`);
```

Comment:
This needs ArrayPrototypeMap or we can remove ObjectEntries. I prefer removing primordials, but I'm just a voice in a desert :-))

Reply:
No you're not. We are at least two. Lol.
Thanks for spotting that, removed.